### ISSUE TITLE
Exception during code highlighting ("Invalid id given")

### ISSUE THREAD

**Issue by iovuio** [asciidoctor/asciidoctor-intellij-plugin#240]
**Steps to reproduce**

I cannot identify the exact steps how to reproduce this.
I was editing the following file: https://github.com/assertj/doc/blob/master/src/docs/asciidoc/user-guide/assertions-guide.adoc

Stacktrace at the end of this report.

**AsciiDoc Plugin Version:** v0.24.3

**IDE Name and Version**: IntelliJ IDEA 2018.3.5 (Ultimate Edition)
Build #IU-183.5912.21, built on February 26, 2019
JRE: 1.8.0_152-release-1343-b28 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.3

**Stacktrace**

```
Error rendering preview: org.jruby.exceptions.ArgumentError: (ArgumentError) Invalid id given: java]* link:#assertj-core-recursive-comparison-comparators[Specifying comparators to use in the comparison

org.asciidoctor.internal.AsciidoctorCoreException: org.jruby.exceptions.ArgumentError: (ArgumentError) Invalid id given: java]* link:#assertj-core-recursive-comparison-comparators[Specifying comparators to use in the comparison
	at org.asciidoctor.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:421)
	at org.asciidoctor.internal.JRubyAsciidoctor.convert(JRubyAsciidoctor.java:384)
	at org.asciidoctor.internal.JRubyAsciidoctor.render(JRubyAsciidoctor.java:210)
	at org.asciidoc.intellij.AsciiDoc.render(AsciiDoc.java:204)
	at org.asciidoc.intellij.editor.AsciiDocPreviewEditor$2.run(AsciiDocPreviewEditor.java:159)
	at org.asciidoc.intellij.editor.LazyApplicationPoolExecutor$1.run(LazyApplicationPoolExecutor.java:59)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:314)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.jruby.exceptions.ArgumentError: (ArgumentError) Invalid id given: java]* link:#assertj-core-recursive-comparison-comparators[Specifying comparators to use in the comparison
	at uri_3a_classloader_3a_.gems.coderay_minus_1_dot_1_dot_0.lib.coderay.helpers.plugin.validate_id(uri:classloader:/gems/coderay-1.1.0/lib/coderay/helpers/plugin.rb:212)
	at uri_3a_classloader_3a_.gems.coderay_minus_1_dot_1_dot_0.lib.coderay.helpers.plugin.make_plugin_hash(uri:classloader:/gems/coderay-1.1.0/lib/coderay/helpers/plugin.rb:174)
	at org.jruby.RubyHash.default(org/jruby/RubyHash.java:757)
	at org.jruby.RubyHash.[](org/jruby/RubyHash.java:1173)
	at uri_3a_classloader_3a_.gems.coderay_minus_1_dot_1_dot_0.lib.coderay.helpers.plugin.[](uri:classloader:/gems/coderay-1.1.0/lib/coderay/helpers/plugin.rb:52)
	at uri_3a_classloader_3a_.gems.coderay_minus_1_dot_1_dot_0.lib.coderay.encoder.encode(uri:classloader:/gems/coderay-1.1.0/lib/coderay/encoder.rb:83)
	at uri_3a_classloader_3a_.gems.coderay_minus_1_dot_1_dot_0.lib.coderay.duo.encode(uri:classloader:/gems/coderay-1.1.0/lib/coderay/duo.rb:66)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.substitutors.highlight_source(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/substitutors.rb:1485)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.substitutors.apply_subs(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/substitutors.rb:119)
	at org.jruby.RubyArray.each(org/jruby/RubyArray.java:1792)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.substitutors.apply_subs(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/substitutors.rb:106)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.block.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/block.rb:116)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.html5.listing(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/html5.rb:632)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.base.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/base.rb:38)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:75)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:84)
	at org.jruby.RubyArray.map(org/jruby/RubyArray.java:2577)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:84)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.html5.section(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/html5.rb:369)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.base.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/base.rb:38)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:75)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:84)
	at org.jruby.RubyArray.map(org/jruby/RubyArray.java:2577)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:84)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.html5.section(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/html5.rb:368)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.base.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/base.rb:38)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:75)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:84)
	at org.jruby.RubyArray.map(org/jruby/RubyArray.java:2577)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.abstract_block.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/abstract_block.rb:84)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.document.content(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/document.rb:1261)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.html5.embedded(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/html5.rb:270)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.converter.base.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/converter/base.rb:38)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.document.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor/document.rb:1190)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_1_dot_5_dot_8.lib.asciidoctor.convert(uri:classloader:/gems/asciidoctor-1.5.8/lib/asciidoctor.rb:1521)
```

**Comment by ahus1** [comment]
Thank you for this report. From the stack trace and error message I was able to distill a minimal example: once the language (provided after the comma) contains for example a blank, CodeRay ejects this with an exception as it is no valid identifier. At least for CodeRay should match [a-z_]

I'll submit this as a question to the Asciidoctor project how this should be handled.

````
[source,j a]
----
anything
----
````

**Comment by iovuio** [comment]
👍 Thank you @ahus1 
It's magic to me how you can get to minimal samples just from those stacktraces .

**Comment by ahus1** [comment]
Thanks to Dan this will be fixed in Asciidoctor 2.0.0. This bug is on hold until the AsciidoctorJ release is available.

**Comment by ahus1** [comment]
This is fixed in Asciidoctor 2. Preview-Builds are available https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases. A final release will be available after AsciidoctorJ is released as a final version.
